presentation/src/main/java/com/sopt/presentation/calendar/calendarPeriod/CalendarPeriodRoute.kt

```diff
@@ -0,0 +1,172 @@
+    onButtonClick: (String, String, Int, Boolean, List<String>) -> Unit,
+    appointmentName: String,
+    category: String,
+    time: Int,
+    days: List<String>
+) {
+    var startDate by remember { mutableStateOf("") }
+    var endDate by remember { mutableStateOf("") }
+    var dates by remember { mutableStateOf(listOf<String>()) }
+    var isSingleDateMode by remember { mutableStateOf(false) }
+
+    val typography = NoostakTheme.typography
+    val colors = NoostakTheme.colors
+
+    Scaffold(
+        modifier = Modifier
+            .statusBarsPadding()
+            .navigationBarsPadding(),
+        topBar = {
+            NoostakTopAppBar(
+                title = stringResource(R.string.text_calendar_appointment),
+                isIconVisible = false
+            )
+        }
+    ) { innerPadding ->
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(innerPadding)
+                .padding(dimensionResource(id = R.dimen.horizontal_padding))
+        ) {
+            Spacer(modifier = Modifier.height(18.dp))
+
+            NoostakProgressBar(progressBar = listOf(false, true, false))
+
+            NoostakHeaderText(text = stringResource(R.string.text_calendar_appointment_choose))
+
+            Row(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(top = 18.dp),
+                verticalAlignment = Alignment.CenterVertically,
+                horizontalArrangement = Arrangement.SpaceBetween
+            ) {
+                Text(
+                    text = stringResource(R.string.text_calendar_appointment_one_day),
+                    style = typography.b2Regular,
+                    textAlign = TextAlign.Start,
+                    color = colors.gray900
+                )
+                NoostakSwitch(
+                    checked = isSingleDateMode,
+                    onCheckedChange = { isSingleDateMode = it }
+                )
+            }
+            Spacer(modifier = Modifier.height(20.dp))
+            Box(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .height(1.dp)
+                    .background(colors.gray200)
+                    .padding(bottom = 17.dp)
+            )
+
+            NoostakCalendar(
+                start = startDate,
+                end = endDate,
+                isSingleDate = isSingleDateMode,
+                isRangeSelected = { selectedDates ->
+                    if (isSingleDateMode) {
+                        dates = selectedDates
+                        startDate = ""
+                        endDate = ""
+                    } else {
+                        dates = selectedDates
+                        if (selectedDates.isNotEmpty()) {
+                            startDate = selectedDates.first()
+                            endDate = selectedDates.last()
+                        }
+                    }
```

Comment:
P1: 어떤 경우이든 selectedDates 리스트에 날짜를 넣고 있기 때문에 공통 로직은 if문 바깥으로 빼주는게 가독성에 좋을 것 같아요!
```suggestion
                   dates = selectedDates
                    if (isSingleDateMode) {
                        startDate = ""
                        endDate = ""
                    } else {
                        if (selectedDates.isNotEmpty()) {
                            startDate = selectedDates.first()
                            endDate = selectedDates.last()
                        }
                    }
```

Reply:
녜